index.html

```diff
@@ -107,11 +112,69 @@ <h3><a href="https://github.com/FIRSTMap/firstmap.github.io">View or edit FIRSTM
             </div>
         </div>
 
+        <div id="search">
+            <h2 class="stripe">Search</h2>
+
+            <div id="searchSection" hidden>
+                <h3>Search</h3>
+                <div id="teamSearch">
+                    <input type="search" id="teamSearchInput" onkeyup="inputKeyUp(this)" />
+                    <button onclick="search()">Search</button>
+                </div>
+
+                <div id="eventSearch" hidden>
+                    <input type="search" id="eventSearchInput" />
+                    
+                    <ul id="eventSearchList">
+                    </ul>
+                </div>
+
+                <select id="searchType" onchange="(() => { updateSearchType(); })()">
+                    <option value="team">Team</option>
+                    <option value="event">Event</option>
+                </select>
+                <br />
+                <h6 id="searchInfoText"></h6>
+            </div>
+
+            <div id="filterSection" hidden>
+                <h3>Filter</h3>
+                <div id="teamFilter">
+                    <input type="search" id="filterInputTeams" onkeyup="inputKeyUp(this)" />
+                    <button onclick="filter()">Filter</button>
+                </div>
+
+                <div id="eventDistrictFilter" hidden>
+                    <input type="search" id="filterInputOther" />
+                    
+                    <ul id="filterInputList">
+                    </ul>
+                </div>
+                
+                <select id="filterType" onchange="(() => { updateFilterType(); clearFilterClick(); })()">
+                    <option value="team">Team</option>
+                    <option value="event">Event</option>
+                    <option value="district">District</option>
+                </select>
+                <br />
+                <button onclick="clearFilterClick()">Clear Filter</button>
+                <h6 id="filterInfoText"></h6>
+                <br />
+                <h6>
+                    Note: toggle keys do not work in this menu (e.g., pressing 't' on the keyboard
+                    will not toggle team marker visibility, but pressing the toggle button at the
+                    top of the map still works).
+                </h6>
+            </div>
+        </div>
+
         <script src="data/teams.js"></script>
         <script src="data/avatars.js"></script>
         <script src="data/custom_icons.js"></script>
         <script src="data/custom_locations.js"></script>
         <script src="tba-getter.js"></script>
+        <script src="search.js"></script>
+        <script src="searchableSelect.js"></script>
```

Comment:
We should standardize naming of scripts as well. We currently are using both underscores and hyphens and this will add camel case.

Reply:
I'll change `searchableSelect.js` to `searchable_select.js` since there are already two other files with underscores. Should I also change `tba-getter.js` to `tba_getter.js` in the same commit?